src/Raven.Client/Documents/Session/Tokens/IntersectToken.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Text;
+
+namespace Raven.Client.Documents.Session.Tokens
+{
+    public class IntersectToken : QueryToken
```

Comment:
@ayende 
I believe query as follow won't be supported:
`WHERE intersect(Name = 'John', Age > 20) OR LastName = 'Smith'`
since it mixes intersect with the additional OR operator
?